### ISSUE TITLE
[Possible BUG] Plex Availability Checker Not Checking

### ISSUE THREAD

**Issue by bromanguydude** [tidusjar/PlexRequests.Net#672]
Possible Bug?

#### Plex Requests.Net Version:
1.9.7

#### Operating System:
Windows 10

#### Mono Version:
N/A

#### Applicable Logs (from `/logs/` directory or the Admin page):

```
Logs go here (Please make sure you remove any personal information from the logs)
```
I dont have any logs as it is set to Error and it is not feeding anything back to me.

#### Reproduction Steps:

This is a fresh install. The only thing I have changed was Plex Avail Checker refresh time to 15min instead of 60mins.

Basically every other hour it will work fine and sync with plex and show everything I have is available on PlexRequests.Net. Then on off hours it will just show nothing has been checked by PlexRequests and I dont have anything available. I dont change any settings it just flip flops back and fourth.

**Comment by chrisausbun** [comment]
Same thing is happening to me as well. I'm also on version 1.9.7 and Windows 10. I'm sorry, I'm kind of new at using custom built apps on git hub, but is there a way I can easily upgrade my PR 1.9.7 to a dev build or import some of the last commits? Thanks.

**Comment by SuperPotatoMen** [comment]
Don't run the dev build at this moment.

**Comment by tidusjar** [comment]
So, what I think is happening is that on your 'off hours' the checker is actually running.

When we start the checker we clear out the previous result and then start fresh, depending how long it takes to cache the content, it might appear that we cannot identify anything yet (until the cacher finishes its run).

Ideally you should only need to availability checker to run every hour minimum. 
But it would be interesting to see how long it runs for on your system.

What you could do is close Plex Requests,
Start it back up and take note of the time.
Go to the scheduled jobs page and it will show the time that the Plex Availability Checker last run.
Wait for it to finish and then the scheduled jobs page will show the latest time that it run, then check the time difference between when you started it and when it finished.

**Comment by bromanguydude** [comment]
Is there a way to keep the content cached even while Plex Checker rescans similar to the Plex Library scanner? It seems that my library is large enough to keep the scanner running for around 10 minutes. When I had the scanner running every 15 mins then the request page would only work for around 20mins on the hour. If I ran every hour then that would mean around 17% of the time my request page is kinda broken for users and they would be able to request items already available.

Also, I woke up to this one scanner error this morning.;

2016/11/14 05:57:54.247 PlexRequests.Services.Jobs.PlexAvailabilityChecker Error: Failed to obtain Plex libraries System.NullReferenceException: Object reference not set to an instance of an object.
   at PlexRequests.Services.Jobs.PlexAvailabilityChecker.CachedLibraries(PlexSettings plexSettings, Boolean setCache) in C:\projects\requestplex\PlexRequests.Services\Jobs\PlexAvailabilityChecker.cs:line 0


Thanks for the help.

-BMGD

**Comment by chrisausbun** [comment]
Looks like you're right @tidusjar . That's exactly what's happening.

**Comment by bromanguydude** [comment]
Yea the issue is still ongoing even with refresh rate set to 60min. :/

**Comment by tidusjar** [comment]
@bromanguydude I think you are having a different issue, since the error is not a good one...
Are you sure your Plex Settings are correct?

Since we are trying to call Plex's API to get the list of Plex Libraries and an exception is being thrown meaning we are not able to fully cache everything.

**Comment by bromanguydude** [comment]
Hostname and port are correct.
I have tried both with and without SSL.
Im going to try turning Advanced and Episode Search off.

I have typed in the token correctly as it is able to pull it successfully. And even give me a green light when I enter a username and password in.

PR.net also connects with plex via green light.

**Comment by bromanguydude** [comment]
Yea I'm still having the issue :/ , every other hour it will show my items available or not. I think the only way to solve this issue would be to make it so once the Plex library is cached then those items are permanently marked available unless deleted from PlexRequests.Net by the admin. This would eliminate it deleting the "Available" marked items during every refresh and would also eliminate any downtime for users with large library's.

**Comment by SuperPotatoMen** [comment]
Master issue is now https://github.com/tidusjar/PlexRequests.Net/issues/679
